src/dwfl_module.cc

```diff
@@ -10,62 +10,75 @@ extern "C" {
 }
 
 namespace ddprof {
-Dwfl_Module *update_module(Dwfl *dwfl, ProcessAddress_t pc, const Dso &dso,
-                           const FileInfoValue &fileInfoValue) {
+DDProfMod update_module(Dwfl *dwfl, ProcessAddress_t pc, const Dso &dso,
+                        const FileInfoValue &fileInfoValue) {
   if (!dwfl)
-    return nullptr;
+    return DDProfMod();
 
   if (fileInfoValue._errored) {
     LG_DBG("DSO Previously errored - mod (%s)", dso._filename.c_str());
-    return nullptr;
+    return DDProfMod();
   }
   // Now that we've confirmed a separate lookup for the DSO based on procfs
   // and/or perf_event_open() "extra" events, we do two things
   // 1. Check that dwfl has a cache for this PID/pc combination
   // 2. Check that the given cache is accurate to the DSO
 
-  Dwfl_Module *mod = dwfl_addrmodule(dwfl, pc);
-  if (mod) {
-    Dwarf_Addr vm_addr;
-    dwfl_module_info(mod, 0, &vm_addr, 0, 0, 0, 0, 0);
-    if (vm_addr != dso._start - dso._pgoff) {
+  DDProfMod ddprof_mod;
+
+  ddprof_mod._mod = dwfl_addrmodule(dwfl, pc);
+  Dwarf_Addr low_addr = 0;
+  Dwarf_Addr high_addr = 0;
+
+  if (ddprof_mod._mod) {
+    const char *main_name = nullptr;
+    dwfl_module_info(ddprof_mod._mod, 0, &ddprof_mod._low_addr,
```

Comment:
Wow, good update.  I wonder what implications there are for populating these fields more fully.  Any thoughts?

Reply:
I did not understand this question fully. Perhaps we can come back to this.